### ISSUE TITLE
Wrap type=Android SearchBar in TouchableNativeFeedback

### ISSUE THREAD

**Issue by Noitidart** [react-native-elements/react-native-elements#2027]
**Is your feature request related to a problem? Please Describe.**
Pressing the buttons in SearchBar on Android with platform=Android is not showing ripple on buttons, it is doing opacity. 

**Describe the solution you'd like**

If Android, and platform=Android on Searchbar, we should wrap the buttons in TouchableNativeFeedback.

**Describe alternatives you've considered**

react-native-paper does it well - https://callstack.github.io/react-native-paper/searchbar.html

However I prefer react-native-elements, because of the loading indicator, which paper lacks.

**Additional context**

**Comment by shubhamkakkar** [comment]
I would like to work on this issue !!

**Comment by iRoachie** [comment]
Sounds good! Should be easy to implement. [Here](https://github.com/react-native-training/react-native-elements/blob/next/src/searchbar/SearchBar-android.js#L27) specifies all the props for the Icon. The Icon component has a [Component](https://react-native-training.github.io/react-native-elements/docs/icon.html#component) prop that can be used

**Comment by ganes1410** [comment]
@iRoachie is anyone working on this?

**Comment by iRoachie** [comment]
@ganes1410  Nope!

**Comment by ganes1410** [comment]
@shubhamkakkar have you started with this?

**Comment by ganes1410** [comment]
> Sounds good! Should be easy to implement. [Here](https://github.com/react-native-training/react-native-elements/blob/next/src/searchbar/SearchBar-android.js#L27) specifies all the props for the Icon. The Icon component has a [Component](https://react-native-training.github.io/react-native-elements/docs/icon.html#component) prop that can be used

Hi @iRoachie , As you mentioned in this comment, I tried this
```
const defaultClearIcon = {
  type: 'material',
  size: 25,
  color: ANDROID_GRAY,
  name: 'clear',
  Component: TouchableNativeFeedback
};
```
But that is not working on the emulator, I think there is some problem with `TouchableNativeFeedback`. How should i proceed with this ?

**Comment by iRoachie** [comment]
Okay this seems to not be as easy as I thought, some changes need to be made to Icon component first.

**Comment by iRoachie** [comment]
Just implemented this in #2124 without having to change the SearchBar. Try it out and lemme know.

```bash
npm i --save https://github.com/react-native-elements/react-native-elements.git\#icon-refactor
```
Will merge on the weekend.
